cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java

```diff
@@ -271,10 +275,22 @@ public static DataprocConf create(Map<String, String> properties, @Nullable SSHP
       projectId = getSystemProjectId();
     }
 
+    String region = getString(properties, "region");
+    if (region == null || AUTO_DETECT.equals(region)) {
+      region = getRegionFromZone(getSystemZone());
```

Comment:
if the zone is given by the user, it seems like this should derive the region from the provided zone and not from the system zone.

Reply:
Make sense. Fixed.